feat(docs): add setup guide for Azahar (#433)
PretendoNetwork/website: docs/en_US/install/azahar.md

Universal Updater is the easiest way to get Artic Setup Tool. Universal Updater also comes highly recommended for updating all of your other homebrew apps, so installing it is highly recommended. To install Universal Updater, follow <a href="https://universal-team.net/projects/universal-updater.html" target="_blank">this guide</a>.

Once you have Universal Updater, open it and tap the search button on the bottom screen. Type "Artic Setup Tool" into the search box, then tap OK.

Tap the "i" icon on the upper-left corner of the bottom screen, and you should see the item page for Artic Setup Tool. Press A to install it.

Select `AzaharArticSetup.3dsx` if you want to launch the setup tool from the Homebrew Launcher, or select `AzaharArticSetup.cia` if you want to launch it from your HOME Menu.

After a short while, you should see a message pop up on the top screen telling you that the tool has been installed. Exit to the HOME Menu and close out Universal Updater.

Artic Setup Tool is now installed. If you selected the `.cia` file, you should see a message that a new application has been added to the HOME Menu. Press A and you'll now have Artic Setup Tool on your HOME Menu.

> ### Direct title install

Reviewer: This should be better clarified that `Using Universal Updater` and `Direct title install` are two alternatives to install Artic Setup Tool. When reading the guide it seemed like it was a second step instead of an option. Or as Jon said reduce this section.

Before starting, power off your console and insert its SD card into your computer.

You may choose between installing Artic Setup Tool to your HOME Menu or to the Homebrew Launcher.

#### Install to HOME Menu

> If you want to launch Artic Setup Tool from your HOME Menu, download `AzaharArticSetup.cia` from the [latest Artic Setup Tool release](https://github.com/azahar-emu/ArticSetupTool/releases/latest) and put it inside the `cias` folder on your SD card. If the `cias` folder doesn't exist on your SD card, create it.

Reviewer 2: I feel like it could be a little more clear to users in the situation where they don't have FBI installed how they can install FBI
Although they're likely to have GodMode9 as part of the 3DS Hacks Guide, it doesn't mean they know how to use it to install FBI's `.cia` if they do not have it, maybe suggest the `.3dsx` version of FBI instead of just saying;

> download the latest release from...

Or potentially add instructions for installing it's `.cia` through GodMode9 or for downloading it through the Universal-Updater

Once what needs screenshots is decided, I'll happily get to work on those, just lemme know when if that's alright!
Also apologies for commenting twice with this since I'm a little dense when it comes to GitHub

Author: I don't know if this is much of a problem, because the hacks.guide goes over installing FBI from the essentials package. It should pretty much come stock on any user's system where they've followed the guide.

Insert your SD card back into your console.

Open FBI. If you do not have FBI, download the latest release from [GitHub](https://github.com/lifehackerhansol/FBI/releases/latest). Select `SD`, then `cias`. Find and select `AzaharArticSetup.cia`. Select either `Install CIA` or `Install and delete CIA`.

Once it has finished installing, press the HOME button and exit FBI. You should see a message that a new application has been added to the HOME Menu. Press A and you'll now have Artic Setup Tool on your HOME Menu.

#### Install to Homebrew Launcher

If you want to launch Artic Setup Tool from the Homebrew Launcher, download `AzaharArticSetup.3dsx` from the [latest Artic Setup Tool release](https://github.com/azahar-emu/ArticSetupTool/releases/latest) and put it inside the `3ds` folder on your SD card.

Insert your SD card back into your console. From here, you can open the Homebrew Launcher and run the Setup Tool.

## Using Artic Setup Tool

<div class="tip">
	ℹ️ Before opening Artic Setup Tool, ensure that your 3DS/2DS is on the same Wi-Fi network as the computer running Azahar.
</div>

> Open Artic Setup Tool on your 3DS/2DS. Once it's open, press A to run the tool. An IP address will be displayed on the top screen.

Reviewer: If the top screen is broken, the user can also see the IP address by opening the rosalina menu, then going to debug options then enabling the debugger. The IP will appear in the bottom screen. May be worth adding.

Open Azahar on your computer. Click `File`, then `Set Up System Files...`

> Enter the IP address you see on the top screen into the text box in Azahar, making sure to include the `:5543` at the end. Click OK.

Reviewer: Not really needed, the default port if it is omitted is already 5543. Also, you need to mention that Old 3DS setup will be selected by default which is normal.

> On Azahar, a screen will appear telling you to "update your Nintendo 3DS system". Click OK to allow the update to proceed.

Reviewer: There is a common error here that can happen if the otp backup is not on the user's SD card. This backup is only created when updating Luma3DS or when the Luma3DS version is lost due to power loss (like removing the battery). Sadly, removing the battery is the only way to regenerate this backup.

Author: If you can tell me exactly what that error says, I'll create a dedicated troubleshooting section at the bottom for it.

Author: Bump. I'd appreciate if you could find that error message for me.

After a few minutes, Azahar will notify you that the update is complete. Click OK to continue. <b>You are not finished with Artic Setup Tool.</b>

> In Azahar, click `File` and `Set Up System Files...` again. Enter the IP address again, and this time, click `New 3DS Setup`. <b>This will work regardless of whether or not your system is a New 3DS/2DS.</b> Click OK.

Reviewer: `This is required regardless of...`

Once again, a screen will appear on Azahar telling you to "update your Nintendo 3DS system". Click OK to allow the update to proceed. Another screen will appear telling you some important information about system updates. Review this information and click "I Accept" to proceed to the update.

Once the update is complete, click OK.

On your 3DS/2DS system, press START to exit Artic Setup Tool. You are now done with your 3DS/2DS system for this setup. It is no longer needed for the rest of this process.

## LLE module setup in Azahar

In Azahar, click `Emulation`, then `Configure`.

> Select the `System` menu and click `Enable required LLE modules for online features`. Make sure this box is checked, then click OK.

Reviewer: We shouldn't be telling users to have this always enabled. Having LLE modules enabled disables savestates (due to some things not fully implemented). This should be made clear here.

Author: I'll make note that enabling this will disable savestates.

## Downloading Nimbus

Download the [latest Nimbus release](https://github.com/PretendoNetwork/Nimbus/releases/latest). Be sure to select a zip file labeled either `cia` or `combined`. Once downloaded, extract this zip file.

> In Azahar, click `File` and `Open Azahar Folder`. A file explorer window will open. Enter the `sdmc` folder and paste the `3ds` and `cias` folders from the zip file here. If everything goes correct, you should have these two folders in `sdmc` alongside a folder called `Nintendo 3DS`.

Reviewer: Users may have set a custom SDMC folder so it may not be in the Azahar folder. Maybe I can add an option to open the SDMC folder.

Author: That would be ideal. Otherwise, I'm not sure what the best way to explain it in the guide would be.

Author: I'm going to resolve this one for now, but let me know if "Open SDMC Folder" gets added to Azahar and I'll change this up.

Reviewer 3: Sorry for being late to the party, had some IRL stuff happen

I think the best solution here may just be to write this as kind of generic? Something like "Open your SDMC folder. By default this can be found by *etc etc etc*"

If a user is changing their SDMC folder, I think we can assume they're smart enough to know how to find that custom location that they set themselves without us holding their hand about it, they just need to know that the SDMC folder is where they need to be. These steps mostly help the standard user, in which case their SDMC folder would be the default location

Though I agree that adding a "open SDMC folder" option in the emu would be the actual best, just don't want that to hold things up. We can always change the guide later when/if it's added

Author: Done. Thanks for the suggestion!

> Close Azahar and re-open it. Click `File` and `Install CIA`. Navigate into the `cias` folder from the zip file and select `Nimbus.cia`. This will install Nimbus to your emulator.

Reviewer: `cias` folder from the zip file? You mean from the sdmc?

Author: I actually probably shouldn't have them copy that folder to the sdmc because there's not really much point in having it there. The user still needs to manually click Install CIA and navigate to the folder, so it doesn't matter where it is.

It would be cool if Azahar had a feature to automatically detect CIAs in a given folder and show them in the title list for the user to install, but I guess that's a separate issue.

## Using Nimbus

> Open Nimbus within Azahar. Upon opening it, it will notify you that it's been updated. Press START on your controller to restart the system, or press M on your keyboard. Nimbus should automatically reopen itself.

Reviewer: I wouldn't say press M as users may have it bound to something else. Does just stopping emulation also work?

Author: I'll test this soon.

Author: From what I can tell, pressing start only just reboots the system (doesn't do anything else at the same time), and therefore stopping emulation should have the exact same effect.

Upon Nimbus reopening, click the Pretendo button. Once you have switched to Pretendo, click `Emulation` and then `Stop` to exit Nimbus.

> 

Reviewer: Maybe add an extra step here to launch the home menu and verify you are properly getting a friend code.

Author: I considered this, but noticed the friend list requires that you have a personal Mii set (which the setup tool doesn't carry over from the user's system), so I figured having the user make a whole Mii is too much of a hassle for a simple test of whether they got a friend code. But if you think it's necessary, I can include it anyway.

Reviewer 3: I wasn't aware that's how this worked to be honest, I was under the impression that this whole process effectively cloned your 3DS to your PC for use in the emulator so your account would also be carried over, like how the Cemu guide does it

That sounds like it would also be a point of confusion for users who would expect this to work the way Cemu does, and not understand why their account hasn't been carried over and such and assume it's a bug or they messed up the guide or something. This might need to be noted somewhere then, so we/Azahar don't get support threads from users asking why their online progress has been """reset""" and such

As for the suggestion itself, I think it's worth having since without a friend code users can't really do anything online. The guide doesn't have to get super detailed about it though, just telling people "check the friends list" should be enough?

Author: Added. Thanks for the suggestions!

Open Azahar on your computer. Click `File`, then `Set Up System Files...`. A window will pop up containing some information and a text box. `Old 3DS Setup` will be selected regardless of what system you have. <b>This is normal.</b> Enter your 3DS/2DS IP address into the text box in Azahar and click OK.

<img src="/assets/images/docs/install/azahar/azahar_set_up_enterIP.png" width=100% height=auto/>

<div class="tip yellow">
	⚠️ If you see an error that says "Missing OTP backup on SD card", you need to remove your 3DS/2DS system's battery and reinsert it to generate the OTP backup.
</div>

<div class="tip yellow">
	⚠️ If you see an error that says "The country configuration does not match the console region", you need to change the country set in Azahar to match the region of your system. To do this, click "Emulation", then "Configure", then click "System". Find the country drop-down box and select your system's country from the list, then click OK to save your configuration settings. <b>Please note that "Country" and "Region" are two separate drop-down boxes in Azahar.</b> "Country" is the one that needs to be changed to fix this issue.
</div>

> On Azahar, a screen will appear telling you to "update your system". This is actually a process that downloads some required system files from Nintendo so that Azahar can connect online. This won't modify anything on your 3DS/2DS system. Click OK to allow it to proceed.

Reviewer: > downloads sime required system files from the 3DS update servers

Open your emulator's SDMC folder. By default, this will be located in the main Azahar folder. To get there, click `File` and `Open Azahar Folder`. A file explorer window will open. Enter the `sdmc` folder and paste the `3ds` folder from the zip file into there. If everything goes correct, inside the `sdmc` folder you should have two folders named `3ds` and `Nintendo 3DS` (with the `Nintendo 3DS` folder already being present).

Close Azahar and re-open it. Click `File` and `Install CIA`. Navigate into the `cias` folder from the zip file and select `nimbus.cia`, then click OK. This will install Nimbus to your emulator.

## Using Nimbus

Open Nimbus within Azahar. Upon opening it, it will notify you that it's been updated. Click `Emulation` and then `Stop` to exit it.

<img src="/assets/images/docs/install/azahar/azahar_nimbus_update.png" width=100% height=auto/>

Reopen Nimbus and click the Pretendo button. Once you have switched to Pretendo, click `Emulation` and then `Stop` to exit Nimbus again.

<img src="/assets/images/docs/install/azahar/azahar_nimbus_pretendo_select.png" width=100% height=auto/>

> Once you're connected to Pretendo Network, you may want to open the HOME Menu and check the Friend List to confirm that you've been assigned a friend code. Please note that Artic Setup Tool does not copy your account or save data from your system. To sync your save data with Azahar, try the <a href="https://db.universal-team.net/3ds/artic-base-server" target="_blank">Artic Base Server</a> homebrew app.

Reviewer: Artic Base is to be able to play the 3DS games on your 3DS directly on Azahar by broadcasting their resources, it's not to sync saves.

Author: The readme says that it syncs saves with the emulator as you play too. I'll remove that if that's not the case though.

# Azahar

<div class="tip red">
	<strong>CAUTION:</strong>

> 	This guide only supports official builds provided by the Azahar emulator team. Any unofficial/third-party emulator builds are not supported, and we do not provide technical support for issues caused by using said unofficial builds.

Reviewer 3: I think this is more correct? Every where I've seen Azahar write their own branding, "emulator" has been capitalized like this

```suggestion
	This guide only supports official builds provided by the Azahar Emulator team. Any unofficial/third-party emulator builds are not supported, and we do not provide technical support for issues caused by using said unofficial builds.
```

</div>

<div class="tip">
	ℹ️ Azahar <b>requires</b> connecting to a real 3DS/2DS system using an app called Artic Setup Tool before you can play online. This guide assumes that you have a <b>Homebrewed System running the latest version of Luma3DS (v13.3.1 or higher)</b>. If you don't, please follow this <a href="https://3ds.hacks.guide/" target="_blank">guide</a> on how to install homebrew to your system first.
</div>

> The following steps are required for you to connect to the Pretendo Network:

Reviewer 3: We don't really refer to ourselves like this in other places, so just a consistency thing

```suggestion
The following steps are required for you to connect to Pretendo Network:
```

If you want to launch Artic Setup Tool from your HOME Menu, download `AzaharArticSetup.cia` from the [latest Artic Setup Tool release](https://github.com/azahar-emu/ArticSetupTool/releases/latest) and put it on your SD card. Then, open FBI and install it.

If you want to launch Artic Setup Tool from the Homebrew Launcher, download `AzaharArticSetup.3dsx` from the [latest Artic Setup Tool release](https://github.com/azahar-emu/ArticSetupTool/releases/latest) and put it inside the `3ds` folder on your SD card.

## Using Artic Setup Tool

<div class="tip">
	ℹ️ Before opening Artic Setup Tool, ensure that your 3DS/2DS is on the same Wi-Fi network as the computer running Azahar.
</div>

Open Artic Setup Tool on your 3DS/2DS. Once it's open, press A to run the tool. An IP address will be displayed on the top screen.

If your system's top screen is broken, hold L + D-Pad Down + SELECT to open the Rosalina menu. Go to `Debugger options...` and enable the debugger, then press B. Your IP address will be displayed in the top-right corner of the bottom screen. Write this IP address somewhere, then disable the debugger and exit the Rosalina menu.

> Open Azahar on your computer. Click `File`, then `Set Up System Files...` A window will pop up containing some information and a text box. `Old 3DS Setup` will be selected regardless of what system you have. This is normal.

Reviewer 3: ```suggestion
Open Azahar on your computer. Click `File`, then `Set Up System Files...`. A window will pop up containing some information and a text box. `Old 3DS Setup` will be selected regardless of what system you have. This is normal.
```

> Enter the IP address you see on the top screen into the text box in Azahar and click OK.

Reviewer 3: Minor correction since the top screen may be broken

```suggestion
Enter your 3DS/2DS IP address into the text box in Azahar and click OK.
```

> On Azahar, a screen will appear telling you to "update your Nintendo 3DS system". Click OK to allow the update to proceed.

Reviewer 3: Is this a real firmware update? And if so, what does it do if you're already on the latest firmware (if it even shows at all in that case)? And if not, we should probably tell people what it actually is, or at the very least that this is normal and to not worry

As a user who has never set up Azahar before, this line is a bit confusing to me, and there's a subset of users who stay on older firmwares out of fear of updating them so they'll probbaly question this too. @PabloMK7 can you clarify?

Author: This process is basically the emulator downloading system files from Nintendo. While it's technically an "update", it's not really updating from any other version since the emulator doesn't have any system files yet.

Reviewer 3: I would probably leave a note then about that, just to not scare/confuse people who may think this is modifying their actual console (like I thought it might be). Doesn't have to be super detailed, just something like "this is installing the required system titles for the emulator, not updating your original console" or something?

Author: Added. Thanks!

> On Azahar, a screen will appear telling you to "update your Nintendo 3DS system". Click OK to allow the update to proceed.

Reviewer 2: An error which may occur at this stage would be if the country setting of the emulator doesn't match that of the console:

> `The country configuration does not match the console region. Please select a valid country from the emulation settings.`

Perhaps add instructions for setting this, as I feel inexperienced users may get a little lost trying to find the country settings for Azahar, or at least a note about it

Author: Does this appear in the setup tool or the emulator and at what point exactly does it appear?

Reviewer 2: The error message appears on the 3DS itself, in Artic Setup Tool, and occurs when trying to connect to the console using Azahar to set up system files

Author: Alright thanks, I've committed a fix.

## Downloading Nimbus

Download the [latest Nimbus release](https://github.com/PretendoNetwork/Nimbus/releases/latest). Be sure to select a zip file labeled either `cia` or `combined`. Once downloaded, extract this zip file.

Open your emulator's SDMC folder. By default, this will be located in the main Azahar folder. To get there, click `File` and `Open Azahar Folder`. A file explorer window will open. Enter the `sdmc` folder and paste the `3ds` folder from the zip file here. If everything goes correct, you should have this folder in `sdmc` alongside a folder called `Nintendo 3DS`.

Close Azahar and re-open it. Click `File` and `Install CIA`. Navigate into the `cias` folder from the zip file and select `nimbus.cia`. This will install Nimbus to your emulator.

## Using Nimbus

Open Nimbus within Azahar. Upon opening it, it will notify you that it's been updated. Click `Emulation` and then `Stop`.

Reopen Nimbus and click the Pretendo button. Once you have switched to Pretendo, click `Emulation` and then `Stop` to exit Nimbus again.

> Once you're connected to Pretendo, you may want to open the HOME Menu and check the Friends List to confirm that you've been assigned a friend code. Please note that Artic Setup Tool does not copy your account or save data from your system. To sync your save data with Azahar, try the <a href="https://db.universal-team.net/3ds/artic-base-server" target="_blank">Artic Base Server</a> homebrew app.

Reviewer 3: When in official capacities we prefer to use the full project name rather than the nickname (I know there's a few places where this isn't the case, those are mistakes that I'm trying to slowly update over time)

This was supposed to be commented at the same time as the approval, not sure why it didn't go through at first. The approval still stands though, this isn't a blocker

```suggestion
Once you're connected to Pretendo Network, you may want to open the HOME Menu and check the Friends List to confirm that you've been assigned a friend code. Please note that Artic Setup Tool does not copy your account or save data from your system. To sync your save data with Azahar, try the <a href="https://db.universal-team.net/3ds/artic-base-server" target="_blank">Artic Base Server</a> homebrew app.
```

Author: Thanks! Adding. I'll also double check the whole thing and make sure there aren't any other instances like this because I think there might be.

Edit: Looks like apart from references to the button in Nimbus (which are correct), there's no others.
